### ISSUE TITLE
False Positive: Lambda is incorrectly flaged by OperatorWrap

### ISSUE THREAD

**Issue by Aman-Baliyan** [checkstyle/checkstyle#20541]
I have read check documentation: https://checkstyle.org/checks/whitespace/operatorwrap.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash

$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
    "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
    "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="OperatorWrap">
      <property name="option" value="nl"/>
      <property name="tokens"
                value="QUESTION, COLON, EQUAL, NOT_EQUAL, DIV, PLUS, MINUS,
            STAR, MOD, SR, BSR, GE, GT, SL, LE, LT, BXOR, BOR, LOR, BAND,
            LAND, LITERAL_INSTANCEOF, TYPE_EXTENSION_AND, ASSIGN, DIV_ASSIGN,
            PLUS_ASSIGN, MINUS_ASSIGN, STAR_ASSIGN, MOD_ASSIGN, SR_ASSIGN, BSR_ASSIGN,
            SL_ASSIGN, BXOR_ASSIGN, BOR_ASSIGN, BAND_ASSIGN, METHOD_REF, LAMBDA"/>
    </module>
  </module>
</module>

$ cat Test.java
package temp;
public class ForTest {
    public void doIllegal() {
        Runnable r = () -> { // violation here
            int a = 1;
        };
    }
}

$ java -jar checkstyle-14.1.0-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/roman/tmp/Test.java:4:25: '->' should be on a new line. [OperatorWrap]
Audit done.
Checkstyle ends with 1 errors.

```

**Describe what you expect in detail.**

Lambda should not flagged because code is wrapped on it, wrapping is after `{` so it should not be violation.
Attention `=` (EQUAL) is in config, but it has no violation on ` int a = 1;` as line has no wrapping.

**Comment by romani** [comment]
@Aman-Baliyan , why you think this should no violation? 

nl demands to be like:
```
 Runnable r = () 
                 -> {  // no violation expected here
            int a = 1;
            int b = 2;
            int c = 3;
            int e = 5;
            int h = 8;
            int i = 9;
            int j = 10;
        };
```

**Comment by Aman-Baliyan** [comment]
@romani in these examples ->

```
int a = 3 + 4
                    + 6;

        int c = a == 0 ? 3
                    : 4;
```
OperatorWrap does not throw any violation for the `+` between 3 and 4 and also no violation is thrown for `?` before 3 then why there is violation for '->'

That's why i thought this is a false positive for lambda . Is this reason is valid or not?

I think OperatorWrap only throws violation when the operator is present at the end of line when config is set to `nl`

**Comment by Aman-Baliyan** [comment]
@romani ping

**Comment by Aman-Baliyan** [comment]
@romani ping

**Comment by Aman-Baliyan** [comment]
Please approve the issue if it is valid.

**Comment by romani** [comment]
should be no violation
